src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs

```diff
@@ -181,18 +180,22 @@ internal static JsonValue CreateFromTypeInfo<T>(T value, JsonTypeInfo<T> jsonTyp
 
         internal static JsonValue? CreateFromElement(ref readonly JsonElement element, JsonNodeOptions? options = null)
         {
-            if (element.ValueKind is JsonValueKind.Null)
+            switch (element.ValueKind)
             {
-                return null;
-            }
+                case JsonValueKind.Null:
+                    return null;
 
-            // Force usage of JsonArray and JsonObject instead of supporting those in an JsonValue.
-            if (element.ValueKind is JsonValueKind.Object or JsonValueKind.Array)
-            {
-                ThrowHelper.ThrowInvalidOperationException_NodeElementCannotBeObjectOrArray();
-            }
+                case JsonValueKind.Object:
+                case JsonValueKind.Array:
+                    // Force usage of JsonArray and JsonObject instead of supporting those in an JsonValue.
+                    ThrowHelper.ThrowInvalidOperationException_NodeElementCannotBeObjectOrArray();
+
+                    // The above method will throw, but the compiler doesn't know that.
+                    return default;
```

Comment:
The comment is redundant since this pattern is fairly pervasive in the codebase:
```suggestion
                    return null;
```